src/github.com/matrix-org/dendrite/clientapi/writers/createroom.go

```diff
@@ -22,6 +25,44 @@ type createRoomRequest struct {
 	RoomAliasName   string                 `json:"room_alias_name"`
 }
 
+func (r createRoomRequest) Validate() *util.JSONResponse {
+	whitespace := "\t\n\x0b\x0c\r " // https://docs.python.org/2/library/string.html#string.whitespace
+	// https://github.com/matrix-org/synapse/blob/v0.19.2/synapse/handlers/room.py#L81
+	if strings.ContainsAny(r.RoomAliasName, whitespace) {
+		return &util.JSONResponse{
+			Code: 400,
+			JSON: jsonerror.BadJSON("room_alias_name cannot contain whitespace"),
+		}
+	}
```

Comment:
Synapse doesn't, but I'm inclined to agree, else it'll break parsing in a Bad Way.